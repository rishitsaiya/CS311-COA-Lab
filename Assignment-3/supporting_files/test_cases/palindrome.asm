.data
a:
	4567654
	.text
main:
	load %x0, $a, %x3
	sub %x7, %x7, %x7
loop:
	divi %x3, 10, %x4
	addi %x31, 0, %x30
	muli %x7, 10, %x7
	add %x7, %x30, %x7
	divi %x3, 10, %x3
	bgt %x3, %x0, loop
	load %x0, $a, %x5
	beq %x5, %x7, palindrome
	sub %x10, %x10, %x10
	subi %x10, 1, %x10
	end
palindrome:
	sub %x10, %x10, %x10
	addi %x10, 1, %x10
	end