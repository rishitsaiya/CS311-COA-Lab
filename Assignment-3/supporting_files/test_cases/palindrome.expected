Hash of the Processor State = 155317940

